main_svc_users реализация users

andreyzavrichko opened the pull request.

andreyzavrichko committed:
main_svc_users реализация users

andreyzavrichko committed:
main_svc_users добавление валидации на существующую почту

andreyzavrichko committed:
main_svc_users добавление валидации минимальную длину имени/почты

andreyzavrichko committed:
main_svc_users фикс (whitespace) EmptyLineSeparator

andreyzavrichko committed:
main_svc_users добавление тестов по user

andreyzavrichko committed:
main_svc_users добавление тестов по user

Alextrusk27 commented on main-service/src/main/resources/application.properties:
```diff
@@ -0,0 +1,8 @@
+
+server.port=8080
+
+ewm.server.url = http://localhost:8080
```
Переменная ewm.server.url нужна будет для подключению к стат серверу через клиент. Порт должен быть 9090.
Она криво названа, я ее потом переназову нормально тут и в клиенте. Можно закоментить пока

Alextrusk27 commented on main-service/src/main/java/ru/practicum/ewm/user/mapper/UserMapper.java:
```diff
@@ -0,0 +1,26 @@
+import ru.practicum.ewm.user.model.User;
+
+@Component
+public class UserMapper {
```
Можешь сделать через mapstruct?

Alextrusk27 commented on docker-compose.yml:
```diff
@@ -34,11 +34,25 @@ services:
     container_name: main-service
```
Запуск main-service должен зависеть от запуска бд

Alextrusk27 commented on main-service/src/main/java/ru/practicum/ewm/user/service/UserService.java:
```diff
@@ -0,0 +1,14 @@
+import java.util.List;
+
+public interface UserService {
+    List<UserDto> getUsers(List<Long> ids, int from, int size);
```
От наставника было требование упаковывать параметры в дто. У методов сервиса должен быть только 1 аргумент. В сложных случаях максимум 2.

Alextrusk27 commented on main-service/src/main/java/ru/practicum/ewm/user/exception/ConflictException.java:
```diff
@@ -0,0 +1,11 @@
+package ru.practicum.ewm.user.exception;
```
думаю, директорию с исключениями лучше вынести на уровень выше, т.к. они будут нужны в других местах. Еще тут нужен обработчик исключений

andreyzavrichko committed:
main_svc_users добавление mapstruct,
закомментирование ewm.server.url,
добавление зависимости от запуска бд,
перенос обработчика ошибок, упаковка параметров в дто

andreyzavrichko committed:
main_svc_users фикс обработки 409
